test/suite/test_import01.py

```diff
@@ -74,18 +74,40 @@ def check(self, uri, keys, values):
             else:
                 self.check_record(uri, keys[i], values[i])
 
-    # The ID and checkpoint information can be different between configs, remove it. Everything else
-    # should be the same.
+    # Compare two given configurations, excluding unique data.
     def config_compare(self, aconf, bconf):
-        a = re.sub('id=\d+,?', '', aconf)
-        a = (re.sub('\w+=\(.*?\)+,?', '', a).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', a))
-        b = re.sub('id=\d+,?', '', bconf)
-        b = (re.sub('\w+=\(.*?\)+,?', '', b).strip(',').split(',') +
-             re.findall('\w+=\(.*?\)+', b))
-        a = [x for x in a if not x.startswith("checkpoint=")]
-        b = [x for x in b if not x.startswith("checkpoint=")]
-        self.assertTrue(sorted(a) == sorted(b))
+        # Retrieve the data that can be compared.
+        stripped_aconf = self.strip_subconfig(aconf)
+        stripped_bconf = self.strip_subconfig(bconf)
+
+        self.assertTrue(sorted(stripped_aconf) == sorted(stripped_bconf))
+
+    # Remove information related to the ID and checkpoints from a config.
+    def strip_subconfig(self, conf):
+        subconfigs = []
+        curr_subconfig = ''
+        depth = 0
+
+        for char in conf:
+            if char == '(':
+                depth += 1
+            elif char == ')':
+                depth -= 1
```

Comment:
Can we put a line break right after this for readability?

Reply:
yep! done :)